### PR TITLE
mgr/dashboard: Small improvements for running teuthology tests

### PR DESCRIPTION
**Pull request by zmc**
Part of this PR addresses the reason why failures described in https://github.com/ceph/ceph/pull/24900#issuecomment-439143240 were missed; the other part should just save a few seconds whenever that test runner is invoked.

### CODE REVIEW

**Comment by zmc** [comment]
test this please

File: src/pybind/mgr/dashboard/run-backend-api-tests.sh [hunk 1]
```diff
@@ -17,6 +17,8 @@ if [[ "$1" = "-h" || "$1" = "--help" ]]; then
 	exit 0
```
Comment (s0nea) [inline src/pybind/mgr/dashboard/run-backend-api-tests.sh]: I can't run the script on my local vstart environment anymore:

```
ceph-dev /ceph/src/pybind/mgr/dashboard # ./run-backend-api-tests.sh 
+ setup_teuthology
++ mktemp -d
+ TEMP_DIR=/tmp/tmp.VvZrbWW7xZ
++ pwd
+ CURR_DIR=/ceph/src/pybind/mgr/dashboard
+ BUILD_DIR=/ceph/src/pybind/mgr/dashboard/../../../../build
+ read -r -d '' TEUTHOLOFY_PY_REQS
```

Any ideas? - Did I do something wrong or is something missing?

Comment (zmc) [reply src/pybind/mgr/dashboard/run-backend-api-tests.sh]: Yeah, I had a stray `-e` there - my fault, sorry. The 'debug: ...' commit may or may not be something we actually want; right now I'm trying to figure out why the ceph-dashboard-pull-requests jenkins job never passes.

**Comment by zmc** [comment]
I dropped the debug commit, so this can be merged if we want

**Comment by s0nea** [comment]
Okay, thank you @zmc !
I'm going to schedule a Teuthology run.

File: src/pybind/mgr/dashboard/run-backend-api-tests.sh (hunk 1 above)
Comment (callithea) [reply src/pybind/mgr/dashboard/run-backend-api-tests.sh]: Hey @zmc there seems to be an issue with running the ``run-tox-mgr-dashboard`` tests when running make check/ceph-dashboard tests based on centos. :(  We're still investigating on that - but if you have an idea how to fix this - feedback/suggestions are much appreciated. :)

**Comment by s0nea** [comment]
I scheduled the tests for your PR twice: http://pulpito.ceph.com/compare/?branch=wip-tdehler-testing-25121&suite=rados:mgr
Unfortunately no attempt was successful but I don't think it's related to your PR. E.g. there is currently an issue with the CentOS deployment: https://tracker.ceph.com/issues/37335

I'm not sure if we should wait for a fix here or merge it anyway.

**Pull request merged**